### PR TITLE
win_useradd: Improve unicode support on Py2

### PR DESCRIPTION
**Pull request by twangboy**
### What does this PR do?
Improves Unicode handling. Improved Unicode handling in Python 2.
Fixes Py3 exeption handling issues (exc is non-iterable)
Uses Google Style Docstrings

### What issues does this PR fix or reference?
https://github.com/saltstack/salt/issues/36011

### Tests written?
No

### CODE REVIEW

**Comment by cachedout** [comment]
@twangboy Please lint this.

File: salt/modules/win_useradd.py [hunk 1]
```diff
@@ -58,11 +62,32 @@
...
+    Args:
+        instr (str): String to convert
+
+    Returns:
+        str: Unicode type string
+    '''
+    if instr is None or isinstance(instr, six.text_type):
+        return instr
+    else:
+        return six.text_type(instr, 'utf-8')
```
Comment (cachedout) [inline salt/modules/win_useradd.py]: How are you accounting for non-UTF-8 char sets?

File: salt/modules/win_useradd.py [hunk 2]
```diff
@@ -168,55 +190,35 @@ def update(name,
...
-        False allows the password to expire.
-
-    :param bool disallow_change_password: True blocks the user from changing
-        the password. False allows the user to change the password.
-
-    :return: True if successful. False is unsuccessful.
-
-    :rtype: bool
+    Args:
+        name (str): The user name to update.
```
Comment (cachedout) [inline salt/modules/win_useradd.py]: Can we get some newlines in between these? This is very hard to read. :]

Comment (twangboy) [inline salt/modules/win_useradd.py]: Done

File: salt/modules/win_useradd.py (hunk 1 above)
Comment (twangboy) [inline salt/modules/win_useradd.py]: utf-8 is required for the NetUser* series of commands used by this module.

**Pull request merged**